### DISCUSSION TITLE
Triplanar Texture Support?

### DISCUSSION THREAD

**Discussion by lunkums** [MenacingMecha/godot-psx-style-demo#10] [Q&A] [answered]
Is there any way to enable triplanar textures for the psx shaders provided in this repo?

**Comment by MenacingMecha** [comment, marked as answer]
Not out of the box, no, but you could easily add it by extending [`psx_base.shaderinc`](https://github.com/MenacingMecha/godot-psx-style-demo/blob/master/shaders/psx_base.gdshaderinc) with an `#ifdef` for triplanar mapping.

AFAIK however, triplanar mapping was not present on the console. As such, I'm not interested in accepting any related PRs.

**Reply by lunkums** [reply]
That makes sense, although I think I was generalizing a solution to a problem I'm trying to solve so maybe there's a different way: the problem is creating a material with a repeating texture that still has the psx shader. In the example demo, the background is a scaled box and in order to match the mesh, the floor material has been given a UV scale of 24 x 24. Is there a way to keep the UV scale at 1 while repeating across the entire surface of the mesh rather than stretching across it?

**Reply by MenacingMecha** [reply]
Not without changing the UVs of the mesh, but this is regular basic 3D.

Triplanar mapping is more for curved terrain, where you want to automatically (but imprecisely) map and blend different textures based on normal angle with respect to curvature. Think a series of rocky hills with grass both on the top and on the ground.
